src/net_processing.h

```diff
@@ -89,7 +103,314 @@ struct Peer {
+         * useful when peer will ask for missing transactions by short IDs
+         * at the end of a reconciliation round.
+         */
+        std::map<uint32_t, uint256> m_local_short_id_mapping;
+
+        /**
+         * A reconciliation round may involve an extension, which is an extra exchange of messages.
+         * Since it may happen after a delay (at least network latency), new transactions may come
+         * during that time. To avoid mixing old and new transactions, those which are subject for
+         * extension of a current reconciliation round are moved to a reconciliation set snapshot
+         * after an initial (non-extended) sketch is sent.
+         * New transactions are kept in the regular reconciliation set.
+         */
+        std::set<uint256> m_local_set_snapshot;
+
+        /**
+         * A reconciliation round may involve an extension, in which case we should remember
+         * a capacity of the sketch sent out initially, so that a sketch extension is of the same size.
+         */
+        uint16_t m_capacity_snapshot{0};
+
+        /**
+         * In a reconciliation round initiated by us, if we asked for an extension, we want to store
+         * the sketch computed/transmitted in the initial step, so that we can use it when
+         * sketch extension arrives.
+         */
+        std::vector<uint8_t> m_remote_sketch_snapshot;
+
+        /**
+        * Reconciliation sketches are computed over short transaction IDs.
+        * Short IDs are salted with a link-specific constant value.
+        */
+        uint32_t ComputeShortID(const uint256 wtxid) const
+        {
+            uint64_t s = SipHashUint256(m_k0, m_k1, wtxid);
+            uint32_t short_txid = 1 + (s & 0xFFFFFFFF);
+            return short_txid;
+        }
+
+        /**
+         * Estimate a capacity of a sketch we will send or use locally (to find set difference)
+         * based on the local set size.
+         */
+        uint16_t EstimateSketchCapacity() const
+        {
+            uint16_t set_size_diff = std::abs(uint16_t(m_local_set.size()) - m_remote_set_size);
+            uint16_t min_size = std::min(uint16_t(m_local_set.size()), m_remote_set_size);
+            uint16_t weighted_min_size = m_remote_q * min_size;
+            uint16_t estimated_diff = 1 + weighted_min_size + set_size_diff;
+            return minisketch_compute_capacity(RECON_FIELD_SIZE, estimated_diff, RECON_FALSE_POSITIVE_COEF);
+        }
+
+        /**
+         * Reconciliation involves computing a space-efficient representation of transaction identifiers (a sketch).
+         * A sketch has a capacity meaning it allows reconciling at most a certain number of elements. (see BIP-330).
+         * Considering whether we are going to send a sketch to a peer or use locally, we estimate the set difference.
+         */
+        Minisketch ComputeSketch(const std::set<uint256> local_set, uint16_t& capacity)
```

Comment:
Inside a structure inside `Peer` feels like the wrong place for a lot of this complex logic. I think ideally, `Peer` would continue to be a struct (i.e. data members only) and the logic would be contained in a separate module.